src/chronos/chronos.py

```diff
@@ -209,6 +209,36 @@ def __init__(self, config: ChronosConfig, model: PreTrainedModel) -> None:
     def device(self):
         return self.model.device
 
+    def encode(
+        self,
+        input_ids: torch.Tensor,
+        attention_mask: torch.Tensor,
+    ):
```

Comment:
Just thinking out loud: we could use this also to optimize "batch sampling" (e.g. sampling 20 at a time, five times to get 100 samples)

Reply:
I'm not sure I get it. You mean "caching" the embeddings?